.ado/jobs/playground.yml

```diff
@@ -20,6 +20,12 @@ parameters:
             BuildConfiguration: Release
             BuildPlatform: x64
             SolutionFile: Playground.sln
+          - Name: X64DebugUniversal
+            BuildConfiguration: Debug
+            BuildPlatform: x64
+            SolutionFile: Playground.sln
+            RunWack: true
+            UploadAppx: true
```

Comment:
sync with @jonthysell and @dannyvv - IIRC we had intentionally limited the amount of combinations in the matrix that we were building so that CI was fast enough

Reply:
I was hoping that execution can get parallelized at the job level and that therefore this doesn't necessarily add to the critical path. @dannyvv can you verify?

Reply:
@asklar @dannyvv Forgot to mention: Debug build slice is needed today because Hermes direct debugging - which is subject of the tests added with this change - only works in debug builds. In agreement with Anandraj, I opened [#10245](https://github.com/microsoft/react-native-windows/issues/10245) to make direct debugging work in release builds. Once that's implemented, we can reduce the build matrix here again.

Reply:
It should not affect critical path unless it takes much longer than the other legs.
These take 10.5 minutes and should have no effect on ciritical path, so it is just a matter of resource utilization... Adding those minutes of CPU time per PR is okay for the benefit it gives.
I would actually recommend adding a few more flavors to CI to ensure on other matrix entries coverage